common/Services/ComputeSystemService.cs

```diff
@@ -61,12 +61,13 @@ public async Task<List<ComputeSystemProviderDetails>> GetComputeSystemProvidersA
                     continue;
                 }
 
-                // If we're looking at the Hyper-V extension and the feature isn't present on the users machine, disable the extension.
-                // This can happen if the user is not on a SKU that supports Hyper-V.
-                if (extension.ExtensionClassId.Equals(CommonConstants.HyperVExtensionClassId, StringComparison.OrdinalIgnoreCase) &&
-                    ManagementInfrastructureHelper.IsWindowsFeatureAvailable(CommonConstants.HyperVWindowsOptionalFeatureName) == FeatureAvailabilityKind.Absent)
+                // If we're looking at an internal extension that relies upon a Windows optional feature being present on the machine. We need to
+                // check if its on the machine first before allowing it to be added to the list of extensions that are retrieved. If the feature is
+                // absent we disable the extension.
+                if (ShouldDisabledExtensionIfFeatureAbsent(extension.ExtensionUniqueId))

```

Comment:
How about moving this to ExtensionService to make it more generic? Combined with moving ShouldDisableExtensionIfFeatureAbsent to ManagementInfrastructureHelper (below) other callers would be able to easily use this functionality. And here we would just call 
 DisableExtensionIfFeatureAbsent(extension.ExtensionUniqueId) or
 DisableExtensionIfFeatureAbsent(extension) if "extension" is better for currency here.

Reply:
sounds good, I went ahead and updated it. Thanks!